helix-core/src/main/java/org/apache/helix/util/StatusUpdateUtil.java

```diff
@@ -555,9 +564,10 @@ void publishStatusUpdateRecord(ZNRecord record, Message message, Level level,
    */
   void publishErrorRecord(ZNRecord record, String instanceName, String updateSubPath,
```

Comment:
Instead of using this approach to define param, let's do this:

create a new function:

void publishErrorRecord(ZNRecord record, String instanceName, String updateSubPath,String updateKey, String sessionId, HelixDataAccessor accessor, boolean isController, boolean logToZK);

Put all existing logic in this one and leverage this boolean flag to check.

For the old API, you can call this by passing 
ERROR_LOG_TO_ZK_ENABLED